pyproject.toml

```
@@ -35,22 +35,22 @@ docs = ["sphinx", "enum-tools", "furo"]
 [tool.poetry.group.dev.dependencies]
 pytest = "^7.2.2"
 black = "^23.1.0"
-poethepoet = "^0.22.0"
+poethepoet = "^0.24.4"
 coverage = "^7.2.1"
 pytest-asyncio = "^0.21.1"
-pylint = "2.17.5"
-setuptools = "^68.0.0"
+pylint = "3.0.3"
+setuptools = "^69.0.3"
 pytest-mock = "^3.6.1"
 pytest-xdist = "^3.2.1"
-pyright = "^1.1.298"
+pyright = "^1.1.338"
```

```suggestion
pyright = "1.1.338"
```
Let's require the exact version until the issue with pyright is fixed.